x/concentrated-liquidity/swaps.go

```diff
@@ -535,7 +522,8 @@ func (k Keeper) calcInAmtGivenOut(
 	}
 
 	// coin amounts require int values
-	// truncate amountIn up to avoid under charging
+	// Round amount in up to avoid under charging the user.
+	// Round amount out down to avoid over charging the pool.
 	amt0 := swapState.amountCalculated.Ceil().TruncateInt()
 	amt1 := desiredTokenOut.Amount.ToDec().Sub(swapState.amountSpecifiedRemaining).TruncateInt()
```

Comment:
Shouldn't this rounding behavior be dependent on `zeroForOne`? i.e. if we're swapping from asset 1 to asset 0 we would expect the opposite rounding behavior right?

Reply:
zeroForOne means that zero is token in and one is token out.

We always want to round up token in and round down token out.

We already guarantee that no matter what the value of zeroForOne is, we always round up token in and round down token out